src/main/java/com/sprint/mission/discodeit/service/basic/BasicChannelService.java

```diff
@@ -1,84 +1,134 @@
-        if(channelRepository.findAll().isEmpty()){
-            throw new NoSuchElementException("등록된 채널이 없습니다.");
+    public ChannelFindResponse find(UUID channelId) {
+        Channel channel = channelRepository.findById(channelId)
+                        .orElseThrow(() -> new NoSuchElementException("Channel with id " + channelId + " not found"));
+
+        Optional<Message> lastMessage = messageRepository.findAll().stream()
+                .filter(msg -> msg.getChannelId().equals(channelId))
+                .max(Comparator.comparing(Message::getCreatedAt));
+
+        Instant lastMessageTime = lastMessage.map(Message::getCreatedAt).orElse(null);
+
+        List<UUID> userIds = new ArrayList<>();
+        if(channel.getType().equals(ChannelType.PRIVATE)){
+            userIds = readStatusRepository.findAll().stream()
+                    .filter(status -> status.getChannelId().equals(channelId))
+                    .map(ReadStatus::getUserId)
+                    .toList();
         }
 
-        return channelRepository.findAll();
+        if(channel.getType() == ChannelType.PRIVATE){
+            return new ChannelFindResponse(channel.getType(),lastMessageTime, userIds);
+        }
+        // Public Channel Response
+        return new ChannelFindResponse(channel.getType(), lastMessageTime, channel.getName(), channel.getDescription());
     }
 
     @Override
-    public List<Channel> findByName(String name) {
-        List<Channel> channels = channelRepository.findAll().stream()
-                .filter(c -> c.getName().equalsIgnoreCase(name))
-                .toList();
-        if(channels.isEmpty()){
-            throw new NoSuchElementException("등록된 채널이 없습니다. : " + name);
-        }
-        return channels;
+    public List<ChannelFindResponse> findAllByUserId(UUID userId) {
+        // PUBLIC 채널 목록은 전체 조회합니다.
+        // Public 채널을 List에 저장
+        List<ChannelFindResponse> channelResponses = channelRepository.findAll().stream()
+                .filter(c -> c.getType().equals(ChannelType.PUBLIC))
+                .map(c -> find(c.getId()))
+                .collect(Collectors.toList());
+
+        // PRIVATE 채널은 조회한 User가 참여한 채널만 조회합니다.
+        // userId를 가진 Channel을 readStatusRepository를 통해 가져와서 List에 저장
+        readStatusRepository.findByUserId(userId).stream()
+                .map(ReadStatus::getChannelId)
+                .map(this::find)
+                .forEach(channelResponses::add);
+
+        return channelResponses;
     }
 
     @Override
-    public Optional<Channel> findById(UUID id) {
-        try{
-            return Optional.ofNullable(channelRepository.findById(id));
-        }catch(NoSuchElementException e){
-            throw new NoSuchElementException("등록된 채널이 없습니다. : " + id);
+    public ChannelUpdateDto update(ChannelUpdateDto request) {
+        // PRIVATE 채널은 수정할 수 없습니다.
+        if(channelRepository.findById(request.id()).get().getType() == ChannelType.PRIVATE){
```

Comment:
channelRepository.findById(id) 를 한번만 호출해서 조금 간단하게 만들 수 있을것 같습니다.

```suggestion
        Channel channel = channelRepository.findById(id).orElseThrow(() -> new NoSuchElementException(..))
        if(channel.getType() == ChannelType.PRIVATE) {
           throw ...
        }
        
        channel 가지고 이후 로직 작성..
```